std_execution.bs

```diff
@@ -3022,3 +3039,152 @@ are all well-formed.
         does not call `std::terminate` if an error occurs after control is returned to the caller, the program is ill-formed with no diagnostic required.
 
     2. Otherwise, `execution::start_detached(execution::then(execution::schedule(sch), f))`.
+
+## Coroutine utilities <b>[execution.coro_utils]</b> ## {#spec-execution.coro_utils}
+
+### `execution::as_awaitable` <b>[execution.coro_utils.as_awaitable]</b> ### {#spec-execution.coro_utils.as_awaitable}
+
+1. `as_awaitable` is used to wrap a sender into an object that is awaitable within a particular coroutine. A sender `s` of type `S` can be made awaitable if and only if:
+
+    - `S` models the concept `typed_sender`, and
+
+    - It passes zero or one values of a unique type to `execution::set_value`, and
+
+    - It passes only objects of type `exception_ptr` to `exception::set_error`, and
+
+    - The promise type of the awaiting coroutine implements a function called `unhandled_done` that takes no arguments and returns a `coroutine_handle<>`.
```

Comment:
Does it _have_ to return a `coroutine_handle<void>` or does it just need to be convertible to `coroutine_handle<void>`?

Reply:
Also - do we need to specify that this is a _member_ function?
Or just that the expression `promise.unhandled_done()` is well-formed and returns `coroutine_handle<P>` for some type `P`?
Do we need to say anything about whether such an expression is required to be noexcept?